### ISSUE TITLE
Spring cleaning

### ISSUE THREAD

**Issue by alanorth** [ilri/rmg-ansible-public#17]
A few spring cleaning things:
- Use ansible debconf module instead of spawning a shell or command to echo some strings (like accepting the Oracle JDK license)
- Verify GPG keys for non-default repos explicitly: https://github.com/arachnafobic/ansible-pg94/blob/master/roles/pgdg/tasks/main.yml#L23
- Delete unused roles (in master we should only have roles we use, as the purpose is to alleviate errors or overhead in commonly used configurations)

**Comment by alanorth** [comment]
Awesome. Did you actually try it yet? :)

**Comment by oguya** [comment]
Yes. It works fine, plus it's also idempotent!

```
TASK: [solr | Accept Oracle license] ****************************************** 
ok: [folding01]
```

**Comment by alanorth** [comment]
You said the magic word: idempotent!

On Thu, Nov 12, 2015 at 4:34 PM, James Oguya notifications@github.com
wrote:

> Yes. It works fine, plus it's also idempotent!
> 
> TASK: [solr | Accept Oracle license] ******************************************
> ok: [folding01]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ilri/rmg-ansible-public/issues/17#issuecomment-156103224
> .

## 

Alan Orth
alan.orth@gmail.com
https://alaninkenya.org
https://mjanja.ch
"In heaven all the interesting people are missing." ―Friedrich Nietzsche
GPG public key ID: 0x8cb0d0acb5cd81ec209c6cdfbd1a0e09c2f836c0

**Comment by oguya** [comment]
:+1:  :wink:

**Comment by oguya** [comment]
I don't think if we can verify GPG keys if we obtained them using apt_key[1] module, you can only do that if you used get_url[2], which we are already doing[3] BTW.
1. https://docs.ansible.com/ansible/apt_key_module.html
2. https://docs.ansible.com/ansible/get_url_module.html
3. https://github.com/ilri/rmg-ansible-public/blob/master/roles/solr/tasks/solr.yml#L4

**Comment by alanorth** [comment]
Crap, I had seen an example of it linked in a tweet as a recommendation by some infosec peeps. But finding old tweets on Twitter is tricky... and I apparently didn't retweet it. Hmmm.

**Comment by oguya** [comment]
We actually need all those 9 roles. Anyway, what role(s) did you have in mind that we can clean up ?
BTW we have a lot of branches that we should clean up and or merge into master.

**Comment by alanorth** [comment]
True, about the roles, it's really just the odf, grafana, and arcgis groups in hosts and the arcgis.yml in the root that we don't need.

**Comment by oguya** [comment]
Merged `update-openssl` & `solr-5.2.x` branches into master.
There are still a few [stale branches](https://github.com/ilri/rmg-ansible-public/branches/stale) that we should review, cleanup & merge into master or delete if we no longer need them.
